### ISSUE TITLE
Layer_by_layer Boosting Tree

### ISSUE THREAD

**Issue by samanemami** [tensorflow/tensorflow#51549]
<emThe issue related to the performance of the [BoostedTree classifier](https://github.com/tensorflow/estimator/blob/781c0d30c6bf100aa174591dd97cb70fc39d294d/tensorflow_estimator/python/estimator/canned/boosted_trees_test.py#L403) in terms of accuracy.</em>

**System information**

- OS Platform and Distribution: Linux Ubuntu 16.04, Windows 10
- TensorFlow installed from (source or binary): pip install TensorFlow
- TensorFlow version (use command below): 2.4.1
- Python version: 3.6

** Describe the model**
BoostedTree classifier is a model introduced by [N Ponomareva, T Colthurst, G Hendry.2017] (https://ieeexplore.ieee.org/abstract/document/8257910).
It is built over the Xgboost idea and learning is done through building one layer of decision tree regressor over N boosting iteration.

**Describe the current behavior**
Build one tree for each boosting epoch

**Describe the expected behavior**
As mentioned in its paper, I want to build one tree for all Gradient Boosting Ensemble, and each boosting iteration would be layers of one tree.

**Standalone code to reproduce the issue**
You may find an example of the model, [here](https://www.tensorflow.org/tutorials/estimator/boosted_trees_model_understanding)

**Comment by mohantym** [comment]
Hi!@samanemami , 
This does not  seem to be  a bug or feature request, for any further queries you may open this issue in TF discuss [forum ](https://discuss.tensorflow.org/)as there is a larger community there.Thank you!

**Comment by samanemami** [comment]
Thank you @mohantym 
But my question is about the library performance, as I could not find any related method in the BoostedTree library in the TensorFlow repository regarding the layer_by_layer model.

**Comment by mohantym** [comment]
Hi @Saduf2019  , Could you please look into this issue?

**Comment by samanemami** [comment]
Hi @Saduf2019 
**More detail about the method I am looking for in the BoostedTree library**
I am looking for the LAYER-BY-LAYER BOOSTING Tree which has explained in SECTION V of the [paper](https://ieeexplore.ieee.org/abstract/document/8257910).

But when I reviewed the (library)[https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/python/estimator/canned/boosted_trees.py#L1933-L2102], I could not find any related method, and it works just like other Gradient boosting models.

**Comment by samanemami** [comment]
@Saduf2019 
This repository(https://git.kot.tools/nk2/syntaxnet_rus/-/blob/caae66a144f1237eb6b5c19fa00c317ca3bed09c/tensorflow/tensorflow/contrib/boosted_trees/examples/mnist.py#L93) is an old lib from tf.BoosteTree, which remains.
This is the exact code which the [paper](https://ieeexplore.ieee.org/abstract/document/8257910) used, I am looking for the update of this code in your GitHub repository which includes all the paper methods.

**Comment by samanemami** [comment]
Hi @ymodak 
May I ask if you found anything?
